### PR TITLE
Nil check

### PR DESCRIPTION
**Pull request by harbu**
Fixed the "segmentation fault on nil arguments" issue by adding a check that the given arguments are really strings. If given something else will raise a TypeError. I also updated RSpec to version 2.

### CODE REVIEW

File: lib/levenshtein.rb [hunk 1]
```diff
@@ -1,12 +1,29 @@
...
+
+    # Safe version of distance, checks that arguments are really strings.
+    def distance(str1, str2)
+      validate(str1)
+      validate(str2)
+      ffi_distance(str1, str2)
+    end
+
+    # Unsafe version. Results in a segmentation fault if passed nils!
+    attach_function :ffi_distance, :levenshtein, [:string, :string], :int
```
Comment (dbalatero) [inline lib/levenshtein.rb:20]: FYI, If you want to make this private, you can do: `private :ffi_distance` after the declaration. Let me know if you want to do this before I merge! Thanks!

**Comment by acanyon** [comment]
@dbalatero @harbu Is there anyway we can get this merged?

**Comment by dbalatero** [comment]
hey! thanks for the nudge - i'll try to get to it tonight.

**Comment by dbalatero** [comment]
merged, releasing a 1.0.3 gem shortly.

**Comment by dbalatero** [comment]
1.0.3 is out!

**Pull request merged**